### PR TITLE
Drop Ruby =< 2.4 support

### PR DESCRIPTION
**Pull request by bastelfreak**
@ekohl we can also directly drop Ruby 2.4 and move to 2.5 or 2.7 if you like.

### CODE REVIEW

**Comment by ekohl** [comment]
If you're dropping <= 2.3, I'd certainly also drop 2.4 now. I'd be tempted to keep 2.5 at least for a bit now and only drop it if you need to for Ruby 3.2 support.

**Comment by bastelfreak** [comment]
@ekohl This now drops Ruby 2.3 and Ruby 2.4

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
 
 jobs:
+  rubocop:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.1"
```
Comment (ekohl) [inline .github/workflows/test.yml:20]: We don't run unit tests on 3.1, so should this stick to 3.0?
```suggestion
          ruby-version: "3.0"
```

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,18 +1,103 @@
...
 # The point is for the user to remove these configuration records
 # one by one as the offenses are removed from the code base.
 # Note that changes in the inspected code, or installation of new
 # versions of RuboCop, may require this file to be generated again.
 
 # Offense count: 1
+# This cop supports safe auto-correction (--auto-correct).
+# Configuration parameters: Include.
+# Include: **/*.gemspec
+Gemspec/DateAssignment:
```
Comment (ekohl) [inline .rubocop_todo.yml:13]: I'd be tempted to just fix this by dropping the `.date = ...` bit in gemspec.

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -1,18 +1,103 @@
...
   Exclude:
     - 'lib/metadata_json_lint.rb'
 
 # Offense count: 1
-# Configuration parameters: Exclude.
-Style/Documentation:
+# Configuration parameters: AllowComments, AllowNil.
+Lint/SuppressedException:
   Exclude:
+    - 'Rakefile'
```
Comment (ekohl) [inline .rubocop_todo.yml:34]: Can you fix this one? It's in the `github_changelog_generator` bit. There's also a redundant cop disable above it. If you add a comment in the `rescue` section that it's optional then you can drop this TODO.

File: .rubocop_todo.yml (hunk 2 above)
Comment (bastelfreak) [reply .rubocop_todo.yml:13]: done

File: .rubocop_todo.yml (hunk 3 above)
Comment (bastelfreak) [reply .rubocop_todo.yml:34]: done

**Pull request merged**